src/kernel/task/task_resource.c

```diff
@@ -98,11 +100,16 @@ static int task_resource_module_init(void) {
 		assert(res->resource_offset);
 		*res->resource_offset = offset;
 
-		assert((offset + res->resource_size) <= TASK_RESOURCE_SIZE);
+		//assert((offset + res->resource_size) <= TASK_RESOURCE_SIZE);
+
 		offset += binalign_bound(res->resource_size, sizeof(void *));
+		/* if we have aligned resource we have to reserve space for alignment*/
+		if(res->resource_align) {
+			offset += binalign_bound(res->resource_align, sizeof(void *));
+		}
 	}
 
-	assert(offset == binalign_bound(TASK_RESOURCE_SIZE, sizeof(void *)));
+	//assert(offset == binalign_bound(TASK_RESOURCE_SIZE, sizeof(void *)));
```

Comment:
TASK_RESOURCE_SIZE is a size of allocation to hold all resources of task. Only `<=` acceptable, and in `<` case it's just a waste of memory. In `>` case (now) it's a bug